### DISCUSSION TITLE
Failed to link authenticator

### DISCUSSION THREAD

**Discussion by mikulicf** [dyc3/steamguard-cli#415] [Q&A]
Have a few accounts that I would like to extract the 2FA secrets from, but cannot get the steamguard to activate, I have tried yesterday, today throughout the day with no success, have tried multiple IP Addresses, multiple machines, multiple times....

I seem to always get the error message, regardless of the account, or the machine/IP Address, I do have the same phone number linked on all of these steam accounts, don't know if that could be an issue.

`INFO Polling for tokens... -- If this takes a long time, try logging in again.
INFO Logged in successfully!
INFO Adding authenticator...
ERROR Failed to link authenticator. Account has not been linked. You are sending too many requests to Steam, and we got rate limited. Wait at least a couple hours and try again.
ERROR You are sending too many requests to Steam, and we got rate limited. Wait at least a couple hours and try again.`

**Comment by dyc3** [comment]
> `You are sending too many requests to Steam, and we got rate limited. Wait at least a couple hours and try again.`

There's nothing I can do about rate limiting. I would say wait at least a day and try again.

**Reply by mikulicf** [reply]
I will try waiting a day or two more, will get back here with the results. 
Do you have any guesses on how I got rate limited, and how I could avoid that in the future?

**Reply by dyc3** [reply]
Valve's rate limiting for the HTTP endpoints used by steamguard-cli seem to be pretty heavy, but I personally haven't had a problem with it when testing it myself. I don't have any evidence to support this but: If you're setting up more than 3-5 unique accounts quickly, that could flag you and those accounts to get rate limited. Unfortunately, the steam APIs don't indicate why you got rate limited or when you can retry in the responses.

**Reply by mikulicf** [reply]
Have been setting up multiple accounts, that could be the issue.
Although I have done the same thing in the past with no issues, maybe something related with the timing, I was setting them up at the moment the steam outage happened for a few minutes yesterday, and tried to get my SMS code 2-3 times, seems that my phone number has actually been rate limited, every account that i previously set up with that phone number is unable to receive codes now, even accounts i try on a different IP address are unable to get the codes.
Seems that there is nothing left to do but wait, hopefully they will unblock me soon 😄.

**Reply by dyc3** [reply]
That seems like it would do it. You actually don't need a phone number linked to the account to set up 2fa (this was required previously, but they've relaxed the requirements a bit), the system will send you codes via email instead.

**Reply by mikulicf** [reply]
Oh, I see, didn't know they made the phone number not required, thanks for the info!
